zip-0321.rst

```diff
@@ -0,0 +1,168 @@
+
+URI Syntax:
+
+.. code-block:: EBNF
+
+  zcashurn        = "zcash:" ( zcashaddress [ "?" zcashparams ] / "?" zcashparams )
+  zcashaddress    = *base58
+  zcashparams     = zcashparam [ "&" zcashparams ]
+  zcashparam      = [ addrparam / amountparam / messageparam / labelparam / reqparam / otherparam ]
+  NONZERO         = %x31-39
+  DIGIT           = %x30-39
+  paramindex      = "-" NONZERO *DIGIT
+  addrparam       = "address" [paramindex] "=" zcashaddress
+  amountparam     = "amount"  [paramindex] "=" *DIGIT [ "." *DIGIT ]
+  labelparam      = "label"   [paramindex] "=" *qchar
+  memoparam       = "memo"    [paramindex] "=" *qchar
+  messageparam    = "message" [paramindex] "=" *qchar
+  paramname       = ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )
+  reqparam        = "req-" paramname [ "=" *qchar ]
+  otherparam      = paramname [ "=" *qchar ]
+  qchar           = unreserved / pct-encoded / allowed-delims / ":" / "@"
+  allowed-delims  = "!" / "$" / "'" / "(" / ")" / "*" / "+" / "," / ";"
+
+Here, ``ALPHA``, ``unreserved`` and ``pct-encoded`` are as defined in
+[#RFC3986]_. "base58" is defined as in [#base58check]_
+   
+A ZIP-321 URI represents a request for the construction of a transaction having
+one or more *payments*. In the case that only a single payment is being
+requested, the recipient address MAY be included in the ``hier-part`` component
+of the RFC 3986 URI; otherwise, multiple recipient addresses can be specified
+using ``addrparam`` parameters with different indices. 
+
+Addresses, amounts, labels, and messages sharing the same ``paramindex`` are
+interpreted to be associated with the same payment for the purposes of payment
+construction. Parameters with an empty ``paramindex`` are associated with a
+payment to the address with empty ``paramindex``.
+
+A URI of the form ``zcash:<address>?...`` MUST be considered equivalent to a
+URI of the form ``zcash:?address=<address>&...`` where ``<address>`` is an
+instance of ``zcashaddress``.
+
+If there are any non-address parameters having a given ``paramindex``, then 
+the URI MUST contain an address parameter having that ``paramindex``.
+
+Transfer amount
+--------------------
+
+If an amount is provided, it MUST be specified in decimal ZEC. All amounts MUST
+contain no commas and use a period (.) as the separating character to separate
+whole numbers and decimal fractions. I.e. ``amount=50.00`` or ``amount=50`` is
+treated as 50 ZEC, and ``amount=50,000.00`` is invalid.
+
+Query Keys
+----------
+
+label
+   Label for an address (e.g. name of receiver)
+
+address
+   Zcash address (shielded or transparent)
+
+memo
+   Contents for the zcash shielded memo field. MUST not exceed 512 bytes.
+   Parsers MUST consider the entire URI invalid if the address associated with
+   the same ``paramindex`` is not a valid shielded address.
+
+message
+   Message that clients can display for the purpose of presenting descriptive
+   information about the transaction to the user.
+
```

Comment:
Add an Examples section here. It should also include invalid examples with the reason(s) why they are invalid.

Reply:
I'll produce these from the reference implementation I'm working on.